js/web/lib/wasm/jsep/webgpu/ops/softmax.ts

```diff
@@ -30,19 +31,35 @@ export interface SoftmaxAttributes extends AttributeWithCacheKey {
   readonly axis: number;
 }
 
-const createSoftmaxProgramInfo = (input: TensorView, attributes: SoftmaxAttributes): ProgramInfo => {
-  const shape = input.dims;
-  const outputSize = ShapeUtil.size(shape);
+const createSoftmaxProgramInfo = (context: ComputeContext, attributes: SoftmaxAttributes) => {
+  const input = context.inputs[0];
+  const inputShape = input.dims;
+  const outputSize = ShapeUtil.size(inputShape);
   const WG = 64;
-  let axis = attributes.axis;
-  if (axis < 0) {
-    axis = shape.length + axis;
-  }
-  if (axis < shape.length - 1) {
-    throw new Error('softmax only supports last axis for now.');
+  const inputRank = inputShape.length;
+  const axis = ShapeUtil.normalizeAxis(attributes.axis, inputRank);
+  const needTranspose = axis < inputShape.length - 1;
+  let transposedInput: TensorView;
+
+  if (needTranspose) {
+    const perm = [];
+    for (let i = 0; i < axis; i++) {
+      perm.push(i);
+    }
+    perm.push(inputRank - 1);
+    for (let i = axis + 1; i < inputRank; i++) {
+      perm.push(i - 1);
+    }
```

Comment:
It seems not correct. I would expect axis is the last dim in perm. But it's not. For example , the original dims index array is [0, 1, 2, 3, 4, 5] while axis = 2. The expected perm will be [0, 1, 5, 3, 4, 2]. But actually get [0, 1, 5, 2, 3, 4].

Reply:
My previous understanding on perm is wrong. I just uploaded another PR to fix it. Thanks!